### PR TITLE
encourage use of rapidsai/notebooks for some examples

### PR DESCRIPTION
**Pull request by jameslamb**
Contributes to #374.

Some just-try-out-RAPIDS-on-a-VM examples encourage using the `rapidsai/docker` images, by SSH-ing into the VM and running something like this:

```shell
docker run --gpus all --rm -it \
    --shm-size=1g --ulimit memlock=-1 \
    -p 8888:8888 -p 8787:8787 -p 8786:8786 \
    rapidsai/base:24.06a-cuda11.8-py3.10
```

And then have language suggesting that both of these will work:

* running an interactive `ipython` shell in a terminal
* running Jupyter Lab and accessing it in a web browser

The `rapidsai/base` image doesn't expose a webserver support the browser usecase, and the `rapidsai/notebook` image launches JupyterLab by default.

This PR proposes:

* removing language from those examples about the `ipython` session in the terminal
* encouraging the use of `rapidsai/notebook`, so access from a web browser works on the first try

## How I tested this

Ran through the steps from https://docs.rapids.ai/deployment/nightly/cloud/aws/ec2/, including running the `cudf/10min` notebook... but using the latest `rapidsai/notebook` Docker image instead of `rapidsai/base`.

## Pages this affects

```shell
git grep -E 'install\-rapids\-with\-docker'
git grep -E 'test\-rapids\-docker'
```

Affects the following:

* https://docs.rapids.ai/deployment/nightly/cloud/aws/ec2/
* https://docs.rapids.ai/deployment/nightly/cloud/azure/azure-vm/
* https://docs.rapids.ai/deployment/nightly/cloud/gcp/compute-engine/
* https://docs.rapids.ai/deployment/nightly/cloud/ibm/virtual-server/
* https://docs.rapids.ai/deployment/nightly/guides/l4-gcp/

## Other Changes

This includes 2 other small, semi-related changes I already had staged when starting on this. I'll leave inline comments explaining them.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -128,10 +128,10 @@ The RAPIDS versions for things like container images and install instructions ar
 ```python
 versions = {
     "stable": {
-        "rapids_container": "nvcr.io/nvidia/rapidsai/rapidsai-core:23.02-cuda11.8-runtime-ubuntu22.04-py3.10",
+        "rapids_container": "nvcr.io/nvidia/rapidsai/base:24.04-cuda11.8-py3.10",
     },
     "nightly": {
-        "rapids_container": "rapidsai/rapidsai-core-nightly:23.04-cuda11.8-runtime-ubuntu22.04-py3.10",
+        "rapids_container": "rapidsai/base:24.06a-cuda11.8-py3.10",
```
Comment (jameslamb) [inline README.md:134]: `rapids/rapidsai-core` and `rapidsai/rapidsai-core-nightly` stopped receiving updates about a year ago:

* https://docs.rapids.ai/notices/rsn0030/
* https://github.com/rapidsai/docker/issues/539

This proposes updating this part of the README to eliminate another reference to the old no-longer-supported images.

*(there are more of these in the repo, but that can be a separate PR)*

File: source/examples/rapids-autoscaling-multi-tenant-kubernetes/notebook.ipynb [hunk 2]
```diff
@@ -222,9 +222,9 @@
     "In order to steam the container image to the GKE nodes our image needs to be stored in [Google Cloud Artifact Registry](https://cloud.google.com/artifact-registry/) in the same region as our cluster.\n",
     "\n",
     "```console\n",
-    "$ docker pull rapidsai/rapidsai-core:23.02-cuda11.8-runtime-ubuntu22.04-py3.10\n",
+    "$ docker pull rapidsai/base:24.04-cuda11.8-py3.10\n",
     "\n",
-    "$ docker tag rapidsai/rapidsai-core:23.02-cuda11.8-runtime-ubuntu22.04-py3.10 REGION-docker.pkg.dev/PROJECT/REPO/IMAGE:TAG\n",
+    "$ docker tag rapidsai/base:24.04-cuda11.8-py3.10 REGION-docker.pkg.dev/PROJECT/REPO/IMAGE:TAG\n",
```
Comment (jameslamb) [inline source/examples/rapids-autoscaling-multi-tenant-kubernetes/notebook.ipynb:227]: Similar to https://github.com/rapidsai/deployment/pull/379/files#r1626435519, this is another oudated image reference I noticed.

Will look for them more systematically in a follow-up PR. But some of the references have been out of date for a while, so aren't critical for the 24.06 release.

**Review by jacobtomlinson — APPROVED** [review]
